src/autopas/containers/verletListsCellBased/verletListsCells/neighborLists/VLCAllCellsNeighborList.h

```diff
@@ -79,6 +80,25 @@ class VLCAllCellsNeighborList : public VLCNeighborListInterface<Particle> {
 
     // Helper function to estimate the number of neighbor lists for one base step
     const auto estimateNumLists = [&](size_t baseCellIndex) {
+      // If the cell is near the end of any dimension we can reduce the estimate to the particles in the base cell,
+      // because all other cells' interactions would be from halos.
+      const auto cellIndex3D = utils::ThreeDimensionalMapping::oneToThreeD(baseCellIndex, cellsPerDim);
+      const auto numTouchedBoundaries = [&]() {
+        int acc = 0;
+        for (size_t i = 0; i < 3; ++i) {
+          // false == 0 ; true == 1
+          acc += static_cast<int>(cellIndex3D[i] == (cellsPerDim[i] - 1));
+        }
+        return acc;
+      }();
+      if (numTouchedBoundaries > 0) {
+        if (useNewton3) {
+          return cells[baseCellIndex].size();
+        } else {
+          // In this case we have to accommodate the lists for the reverse interactions from all non-halo neighbors.
+          return cells[baseCellIndex].size() * (3 - numTouchedBoundaries);
```

Comment:
I don't really understand this formula. Maybe explain why this is a suitable estimate (where does the 3 come from)

Reply:
here is a bit of an explanation: e844a86d722a9e9fda3a1790422983ca41af54a5